react/components/atoms/switcher/switcher.js

```diff
@@ -0,0 +1,160 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, Text, Animated, View, Platform } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles } from "../../../util";
+
+import { Touchable } from "../touchable";
+
+export class Switcher extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            checked: PropTypes.bool,
+            disabled: PropTypes.bool,
+            checkedText: PropTypes.string,
+            uncheckedText: PropTypes.string,
+            variant: PropTypes.string,
+            animationDuration: PropTypes.number,
+            marginLeftValue: PropTypes.number,
+            colorInputRangeValue: PropTypes.number,
+            onValueUpdate: PropTypes.func
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            checked: false,
+            disabled: false,
+            variant: "colored",
+            checkedText: null,
+            uncheckedText: null,
+            animationDuration: 200,
+            marginLeftValue: 20,
+            colorInputRangeValue: 150,
+            onValueUpdate: () => {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            checkedData: this.props.checked,
+            marginLeft: this.props.checked
+                ? new Animated.Value(this.props.marginLeftValue)
+                : new Animated.Value(0),

```

Comment:
A small refactor can be done here

```suggestion
            marginLeft: new Animated.Value(this.props.checked ? this.props.marginLeftValue : 0);
```